internal/service/ipam/api_fixed_address_resource_test.go

```diff
@@ -0,0 +1,545 @@
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+
+	"github.com/infobloxopen/bloxone-go-client/ipam"
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/acctest"
+)
+
+/*
+TODO:
+// Add unit test for dhcp_options
+// Add unit test for NextAvailableIP
+// Add unit tests for inheritance
+*/
+
+func TestAccFixedAddressResource_basic(t *testing.T) {
+	var resourceName = "bloxone_dhcp_fixed_address.test"
+	var v ipam.IpamsvcFixedAddress
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccFixedAddressBasicConfig("10.0.0.10", "mac", "aa:aa:aa:aa:aa:aa"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFixedAddressExists(context.Background(), resourceName, &v),
+					// TODO: check and validate these
+					resource.TestCheckResourceAttr(resourceName, "address", "10.0.0.10"),
+					resource.TestCheckResourceAttr(resourceName, "match_type", "mac"),
+					resource.TestCheckResourceAttr(resourceName, "match_value", "aa:aa:aa:aa:aa:aa"),
+					resource.TestCheckResourceAttrPair(resourceName, "ip_space", "bloxone_ipam_ip_space.test", "id"),
+					// Test Read Only fields
+					resource.TestCheckResourceAttrSet(resourceName, "created_at"),
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					//resource.TestCheckResourceAttrSet(resourceName, "inheritance_assigned_hosts"),
+					resource.TestCheckResourceAttrSet(resourceName, "updated_at"),
+					// Test fields with default value
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccFixedAddressResource_disappears(t *testing.T) {
+	resourceName := "bloxone_dhcp_fixed_address.test"
+	var v ipam.IpamsvcFixedAddress
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		CheckDestroy:             testAccCheckFixedAddressDestroy(context.Background(), &v),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccFixedAddressBasicConfig("10.0.0.10", "mac", "aa:aa:aa:aa:aa:aa"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFixedAddressExists(context.Background(), resourceName, &v),
+					testAccCheckFixedAddressDisappears(context.Background(), &v),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccFixedAddressResource_Address(t *testing.T) {
+	var resourceName = "bloxone_dhcp_fixed_address.test_address"
+	var v ipam.IpamsvcFixedAddress
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccFixedAddressAddress("10.0.0.10", "mac", "aa:aa:aa:aa:aa:aa"),
```

Comment:
for this test you should have mac as static, and not changed on update. Only address should be updated.